File: AI-Coding-IDE-main/Backend/App/models/project.py
from datetime import datetime

from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    DateTime,
    ForeignKey
)

from App.database.database import Base


class Project(Base):

    __tablename__ = "projects"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False
    )

    name = Column(
        String(255),
        nullable=False
    )

    description = Column(
        Text,
        nullable=True
    )

    language = Column(
        String(50),
        nullable=False
    )

    created_at = Column(
        DateTime,
        default=datetime.utcnow
    )

    updated_at = Column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow
    )

File: Frontend/src/types/project.ts
export interface Project {
    id: number;
    user_id: number;
    name: string;
    description: string | null;
    language: string | null;
    created_at: string;
    updated_at: string;
}


export interface ProjectCreate {
    name: string;
    description?: string;
    language?: string;
}